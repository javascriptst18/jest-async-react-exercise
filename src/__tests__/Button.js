import React from 'react';
import toJSON from 'enzyme-to-json';
import { mount, render, shallow } from 'enzyme';
import Button from '../components/Button';

it('button state should be enabled from start', () => {
  
})

it('button state should be disabled on click, check state', () => {
  
})

it('button style should be disabled on click, check style', () => {
  
})

/**
 * Snapshot testing is good for smaller components and 
 * making sure the correct styling or class is applied
 * but we have to check the snapshots manually.
 */
it('should match snapshot', () => {
  const wrapper = shallow(<Button onClick={() => {}}/>);
  expect(toJSON(wrapper)).toMatchSnapshot();
})

it('button state should be disabled on click, check state', () => {
   
})