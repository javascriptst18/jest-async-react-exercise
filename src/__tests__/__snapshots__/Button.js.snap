// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<button
  className="bg-pink hover:bg-pink-light 
  text-white 
  font-bold py-2 px-4 
  border-b-4 
  border-pink-dark
  hover:border-pink rounded mt-4
"
  data-test="button"
  onClick={[Function]}
>
  Update Rates
</button>
`;
